### PR TITLE
Added OAuth sign in with LinkedIn

### PR DESCRIPTION
**Pull request by PradyumnaShome**
Addresses #35.

- [x] Added Linkedin as an oauth provider
    - [x] Added a `linkedin_service.go` file mirroring the style of the `github_service.go` file
    - [x] Added `linkedin` to the switch cases in `oauth_service.go`
    - [x] Added `LinkedinUserInfo` and `LinkedInOauthToken` models

### CODE REVIEW

File: service/linkedin_service.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+)
+
+const LINKEDIN_USER_INFO_URL = "https://api.linkedin.com/v1/people/~:(id,formatted-name,email-address,first-name,last-name)"
+
+/*
+	Uses a valid OAuth token to get the user's primary email.
+*/
+func GetLinkedinEmail(oauth_token string) (string, error) {
+	request, err := grequests.Get(LINKEDIN_USER_INFO_URL, &grequests.RequestOptions{
+		Headers: map[string]string{"Authorization": fmt.Sprintf("Bearer %v", oauth_token),
```
[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply service/linkedin_service.go]: I have run `go fmt` multiple times. I don't think it takes extra linebreaks into consideration, but sure I can do it manually.

File: controller/controller.go [hunk 2]
```diff
@@ -23,43 +23,43 @@ func SetupController(route *mux.Route) {
...
+	client_application_url := r.URL.Query().Get("redirect_uri")
 
-	if redirect_uri == "" {
-		redirect_uri = config.AUTH_REDIRECT_URI
+	if client_application_url == "" {
+		client_application_url = config.AUTH_REDIRECT_URI
 	}
 
-	oauth_token, err := service.GetOauthToken(oauth_code.Code, provider, redirect_uri)
+	oauth_token, err := service.GetOauthToken(oauth_code.Code, provider, client_application_url)
```
[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply controller/controller.go]: I found the `redirect_uri` and `redirect_url` confusing (i and l look very similar), and I think the renaming helps you more easily figure out which redirect does what.

[the remark this answers is not in the document]
Comment (PradyumnaShome) [reply controller/controller.go]: The capitalization is just me being pedantic, and trying to differentiate the Authorization Code from the Access Token.

File: service/linkedin_service.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+	if err != nil {
+		return "", err
+	}
+
+	if user_info.ID == "" {
+		return "", errors.New("Invalid OAuth token.")
+	}
+
+	return fmt.Sprintf("Linkedin%v", user_info.ID), nil
```
Comment (ASankaran) [inline service/linkedin_service.go]: `linkedin` instead of `Linkedin`

**Pull request merged**